pkg/handlers/defaulthandler.go

```diff
@@ -0,0 +1,226 @@
+// The function returns a channel that will receive the extracted data bytes and an error if the initial setup fails.
+func (h *DefaultHandler) HandleFile(ctx logContext.Context, input io.Reader) (chan []byte, error) {
+	if h.skipArchives {
+		return nil, nil
+	}
+
+	archiveChan := make(chan []byte, defaultBufferSize)
+	go func() {
+		ctx, cancel := logContext.WithTimeout(ctx, maxTimeout)
+		defer cancel()
+		defer close(archiveChan)
+
+		if err := h.openArchive(ctx, 0, input, archiveChan); err != nil {
+			ctx.Logger().Error(err, "error unarchiving chunk.")
+		}
+	}()
+	return archiveChan, nil
+}
+
+var ErrMaxDepthReached = errors.New("max archive depth reached")
+
+// openArchive recursively extracts content from an archive up to a maximum depth, handling nested archives if necessary.
+// It takes a reader from which it attempts to identify and process the archive format. Depending on the archive type,
+// it either decompresses or extracts the contents directly, sending data to the provided channel.
+// Returns an error if the archive cannot be processed due to issues like exceeding maximum depth or unsupported formats.
+func (h *DefaultHandler) openArchive(ctx logContext.Context, depth int, reader io.Reader, archiveChan chan []byte) error {
+	if common.IsDone(ctx) {
+		return ctx.Err()
+	}
+
+	if depth >= maxDepth {
+		return ErrMaxDepthReached
+	}
+
+	format, arReader, err := archiver.Identify("", reader)
+	if errors.Is(err, archiver.ErrNoMatch) && depth > 0 {
+		return h.handleNonArchiveContent(ctx, arReader, archiveChan)
+	}
+
+	if err != nil {
+		return fmt.Errorf("error identifying archive: %w", err)
+	}
+
+	switch archive := format.(type) {
+	case archiver.Decompressor:
+		// Decompress tha archive and feed the decompressed data back into the archive handler to extract any nested archives.
+		compReader, err := archive.OpenReader(arReader)
+		if err != nil {
+			return err
+		}
+
+		defer compReader.Close()
+
+		return h.openArchive(ctx, depth+1, compReader, archiveChan)
+	case archiver.Extractor:
+		return archive.Extract(logContext.WithValue(ctx, depthKey, depth+1), arReader, nil, h.extractorHandler(archiveChan))
+	default:
+		return fmt.Errorf("unknown archive type: %s", format.Name())
+	}
+}
+
+// extractorHandler creates a closure that handles individual files extracted by an archiver.
+// It logs the extraction, checks for cancellation, and decides whether to skip the file based on its name or type,
+// particularly for binary files if configured to skip. If the file is not skipped, it recursively calls openArchive
+// to handle nested archives or to continue processing based on the file's content and depth in the archive structure.
+func (h *DefaultHandler) extractorHandler(archiveChan chan []byte) func(context.Context, archiver.File) error {
+	return func(ctx context.Context, f archiver.File) error {
+		lCtx := logContext.AddLogger(ctx)
+		lCtx.Logger().V(5).Info("Handling extracted file.", "filename", f.Name())
```

Comment:
Since `filename` is repeated, what about adding it as a value to the context?

(Pseudo-code, may not compile)
```suggestion
		lCtx := logContext.WithValues(logContext.AddLogger(ctx), "filename", f.Name())
		lCtx.Logger().V(5).Info("Handling extracted file.")
```